### ISSUE TITLE
Axios processResponse code failing

### ISSUE THREAD

**Issue by Eltargrim** [RicoSuter/NSwag#1990]
This is my first attempt at using NSwag and I realize that Axios is in preview at this stage. I wanted to provide feedback in case this is an issue with the generation but if the issue is on my end I would appreciate any help resolving.

My generated code includes a get function as well as a process function for the returned object:

```
protected processGetFoo(response: AxiosResponse): Promise<NameValueOfString[]> {
        const status = response.status;
        let _headers: any = {}; 
        if (response.headers && response.headers.forEach) { 
            response.headers.forEach((v: any, k: any) => _headers[k] = v);
        };
        if (status === 200) {
            const _responseText = response.data;
            let result200: any = null;
            let resultData200  = _responseText;
            if (resultData200 && resultData200.constructor === Array) {
                result200 = [] as any;
                for (let item of resultData200)
                    result200!.push(NameValueOfString.fromJS(item));
            }
            return result200;
        } else if (status !== 200 && status !== 204) {
            const _responseText = response.data;
            return throwException("An unexpected server error occurred.", status, _responseText, _headers);
        }
        return Promise.resolve<NameValueOfString[]>(<any>null);
    }
```

The problem with this code is in the `let resultData200 = _responseText;` The array of data is actually in `_responseText.result;`. Because of this it fails the following "if" check and falls through returning null when the promise is resolved. I realize I can fix this manually in every location but obviously that would need to be done every time I regenerate so I'm hoping for a more permanent solution.

Let me know if any additional information is required, or if anyone has a workaround or configuration tweak that can be made to solve this.

Much appreciated!

**Comment by RSuter** [comment]
Issues like that are expected in a new template...

I think this has to be fixed here: 
https://github.com/RSuter/NSwag/blob/master/src/NSwag.CodeGeneration.TypeScript/Templates/AxiosClient.liquid

Or in any of the templates of this PR: 
https://github.com/RSuter/NSwag/pull/1945

**Comment by thijscrombeen** [comment]
@Eltargrim Could you give me your swagger json document so I can reproduce this?

Thanks

**Comment by Eltargrim** [comment]
Here you are @thijscrombeen. I appreciate the help. 

While wholly unfamiliar with this code base it seems to me that we need to utilize the jsonreviver in cases like these? I took a look at this myself over the weekend and I did notice a UseDto flag in the template but I wasn't sure where that should be set in order to try that option for generation.

[swagger.txt](https://github.com/RSuter/NSwag/files/2926783/swagger.txt)

**Comment by thijscrombeen** [comment]
@Eltargrim I've tried to reproduce the issue you are having but so far am unable to. Could you also include the nswag file (your configuration settings basically)?

What version of Axios are you using? 

Thanks

**Comment by Eltargrim** [comment]
Yeah - for sure.

Axios version 0.18.0

configuration attached.
[nswag.txt](https://github.com/RSuter/NSwag/files/2931799/nswag.txt)

There are multiple calls that I have this issue with but one example is the 'getCountries' endpoint.

As I stated in my original post, this is my first attempt at using nswag, so if this can be resolved with configuration changes all the better IMO.

**Comment by thijscrombeen** [comment]
What kind of object does getCountries return?
Is it an array? Is it a dictionary?

`"NameValue[String]": {
            "type": "object",
            "properties": {
                "name": {
                    "type": "string"
                },
                "value": {
                    "type": "string"
                }
            }
        },`
`"/api/services/app/DemoUiComponents/GetCountries": {
            "get": {
                "tags": ["DemoUiComponents"],
                "operationId": "GetCountries",
                "consumes": [],
                "produces": ["text/plain", "application/json", "text/json"],
                "parameters": [{
                    "name": "searchTerm",
                    "in": "query",
                    "required": false,
                    "type": "string"
                }],
                "responses": {
                    "200": {
                        "description": "Success",
                        "schema": {
                            "uniqueItems": false,
                            "type": "array",
                            "items": {
                                "$ref": "#/definitions/NameValue[String]"
                            }
                        }
                    }
                }
            }
        },`

**Comment by thijscrombeen** [comment]
I have created a branch where I made a vue client with axios.
You can find it here: https://github.com/thijscrombeen/NSwag/tree/bug/axios-result-array 

Can you please fork that branch and add an api call similar to getCountries and then generate the axios client so that I can see the end result?

**Comment by Eltargrim** [comment]
Thanks @thijscrombeen - I downloaded the code and began to create an example and it doesn't replicate the issue but I learned a few things in doing so.

The response object in my main project is not defined as part of the swagger.json. The JSON output for the sample endpoint I created is nearly identical to that of my main project. However the API in the same directly returns the NameValue array and in my main project it is wrapped with a response object containing result, exceptions, success boolean, etc.

This brings me back to the fact that perhaps this is not a bug but rather a configuration on my end or potentially a feature request... Is there a way for me to define my response object wrapper in the nswag configuration in some way?

I should also add that Angular generation works for this, with the same configuration (except for the template selection change) but it is handled like this in the generated code:

`let resultData200 = _responseText === "" ? null : JSON.parse(_responseText, this.jsonParseReviver);`

I guess please confirm that this isn't a configuration issue on my end, where I could define the response object that's wrapping my response explicitly for nswag. If that's not possible and we need to get the Axios template to handle this scenario like the Angular template does I'll put some more time in to the sample project for you to replicate the issue I'm running into.

**Comment by thijscrombeen** [comment]
If you could check in a branch with your api code then I could take a look.
As far as I can see now this isn't anything you can make work by changing some configuration.
With your api code I can check also how angular does it and see if we can apply it to axios aswell.

**Comment by Eltargrim** [comment]
Since its been way longer than I intended I just wanted to drop a line and say I haven't abandoned this. I've been pressed with work deadlines for a bit. I still have it on my list to get you a minimum reproducible example. Sorry for the delay.

**Comment by Arsync** [comment]
#2199 can shed a light to the problem. In short say, `resultData200.constructor === Array` must be `Array.isArray(resultData200)`. And nothing else matters...

**Comment by RicoSuter** [comment]
So is this issue fixed with #2199 (constructor => Array.isArray)?

**Comment by RicoSuter** [comment]
Please check the PR.
